pkg/sql/schemachanger/scpb/elements.proto

```diff
@@ -643,3 +645,12 @@ message FunctionParamDefaultExpression {
   uint32 ordinal = 2;
   Expression embedded_expr = 3 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
 }
+
+message DefaultUserPrivilege {
+  uint32 object_type = 1 [(gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/privilege.TargetObjectType"];
+  oneof for_role {
+    cockroach.sql.sqlbase.DefaultPrivilegesForRole.ExplicitRole for_explicit_role = 2;
+    cockroach.sql.sqlbase.DefaultPrivilegesForRole.ForAllRolesPseudoRole for_all_role = 3;
+  }
+  UserPrivileges user_privileges = 4 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+}
```

Comment:
I don't know about this. I feel that it's too early to introduce this element because it's unclear how to model it correctly. Also, we don't actually define any schema changes on it, it's just used for the builder. So, how about this for the time being: add a method somewhere in the scbuildstmt dependencies that returns the default privileges given an ID and whatever else it needs. You can plumb that to the `descCache` in the `builderState` somehow.

Reply:
For a start it's not clear how to key this element, nor whether this element should be exploded into several or not (one for explicit roles, another for all roles, etc on many dimensions). Should all privileges be bundled together?

Consider:
```
BEGIN;
ALTER DEFAULT PRIVILEGES FOR ALL ROLES IN SCHEMA public GRANT INSERT ON TABLES TO CURRENT_USER;
ALTER DEFAULT PRIVILEGES FOR ALL ROLES IN SCHEMA public GRANT DELETE ON TABLES TO CURRENT_USER;
COMMIT;
```
What would the targets look like?

It's not obvious. Until we figure this kind of thing out and exercise it in prod, let's stick to exposing this as dependencies.

As a meta-commentary I'd like to stress that this further reinforces my strong belief that modelling elements should only be done once there's a comprehensive set of schema changes operating on them, and not sooner.

Reply:
I added a `BuildUserPrivilegesFromDefaultPrivileges` which generates the new owner and user privileges elements from db and schema's default privileges. I also deleted all the code I added to reason privileges from default privilege elements (which are not gone) since we can just reuse the interesting function in `catprivilege`.